portal-kernel/src/com/liferay/portal/kernel/portlet/LiferayClientDataRequest.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.kernel.portlet;
+
+import aQute.bnd.annotation.ProviderType;
+
+import javax.portlet.ClientDataRequest;
+
+/**
+ * @author Neil Griffin
+ */
+@ProviderType
+public interface LiferayClientDataRequest
```

Comment:
@ngriffin7a what's the point of this interface? I don't see anyone reference it as a type, only classes implement it. If so, why not just make those classes implement "LiferayPortletRequest, ClientDataRequest" directly?

Reply:
@shuyangzhou I basically created a 1::1 mapping of the Portlet API request/response interface, for the sake of consistency. I will remove it from the interface hierarchy if you prefer.

Reply:
@ngriffin7a please remove remove all non-directly used types, they can be added later without causing major version bump, but not removed without it. So we always prefer deferring api adding to the point it is really needed.

Reply:
@shuyangzhou OK I will remove all non-directly used types locally in preparation for the next PR.

Reply:
@shuyangzhou Thank you so much for doing this review because Preston mentioned that Brian said there is only 1 more day to get stuff like this merged.

Reply:
It is ok @ngriffin7a , we can ask Brian to hold off kernel release for 1 more day, if really necessary.
How many more pulls to go? And please send me an update version of this pull with the while(true) bug fix and the removal for the non-directly used interfaces.

Reply:
@shuyangzhou There are two pulls remaining (this one for https://issues.liferay.com/browse/LPS-80967) and another one for https://issues.liferay.com/browse/LPS-81855 but I'm roadblocked on that until Brian merges https://github.com/brianchandotcom/liferay-portal/pull/60057)

Reply:
Ok, @ngriffin7a that's not too bad, most likely we can get them in tomorrow morning. I will ask Brian to hold off the kernel release to tomorrow afternoon after your 2 pulls are in.